File: programming/2.1/main.cpp
#include <iostream>
#include <cmath>


using namespace std;

float f1(int x) {
  return 3 + x / (x + 3);
}

float f2(int x) {
  return pow(2 - x, 3) + pow(x, -4) + 3 * pow(x, 1/2);
}

float f3(int x) {
  return pow(sin(x), 2) + 25 * cos(x) - 2 / (pow(x, -3) + 2);
}

int main() {
  int x;
  float result;

  cout << "Enter x: ";
  cin >> x;

  result = f1(x) - 2 * f2(x) / f3(x);

  cout << "Result: " << result;
  
  return 0;
}

File: programming/2.1/main.py
from math import sqrt, sin, cos
from random import randint


x = randint(0, 10)

def f1(x: int) -> float:
  return 3 + x / (x + 3)

def f2(x: int) -> float:
  return (2 - x)**3 + x**-4 + 3 * x**1/2

def f3(x: int) -> float:
  return sin(x)**2 + 45 * cos(x) - 2 / (x**3 + 2)

result = f1(x) - 2 * f2(x) / f3(x)

print("Result: ", result)
